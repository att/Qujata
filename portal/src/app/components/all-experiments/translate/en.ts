export const ALL_EXPERIMENTS_TABLE_EN = {
  TITLE: 'All Experiments',
  TABLE_COLUMNS: {
    EXPERIMENT_NAME: {
      NAME: 'Experiment Name',
      ID: 'experimentName'
    },
    ALGORITHMS: {
      NAME: 'Algorithms',
      ID: 'algorithms'
    },
    ITERATIONS: {
      NAME: 'Iterations',
      ID: 'iterations'
    },
    MESSAGE_SIZE: {
      NAME: 'Message Size (KB)',
      ID: 'messageSizes'
    },
    DATE: {
      NAME: 'Date',
      ID: 'date'
    },
    LINKS: {
      DUPLICATE: 'duplicate',
    }
  }
}
