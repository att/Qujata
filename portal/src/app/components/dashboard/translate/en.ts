export const DASHBOARD_EN = {
    CSV_REPORT: {
        TABLE: {
            COLUMNS: {
                ALGORITHM: 'Algorithm',
                SESSION_TIME: 'Session Time - Avg',
                SESSION_HANDSHAKE_TIME: 'Session Handshake Time - Avg',
                DOWNLOAD_SPEED: 'Download Speed - Avg',
            },
        },
    },
};
  