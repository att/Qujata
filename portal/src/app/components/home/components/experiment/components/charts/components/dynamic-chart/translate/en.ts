export const DYNAMIC_CHART_EN = {
    SELECTORS: {
        LABELS: {
            Y_AXIOS: 'Y:',
            X_AXIOS: 'X:',
        },
        PLACEHOLDERS: {
            Y_AXIOS: 'Select Y axios',
            X_AXIOS: 'Select X axios',
            CHART_TYPE: 'Select chart type',
        },
    },
    X_VALUES_TITLE: {
        ITERATIONS: 'Iterations',
    }
};
