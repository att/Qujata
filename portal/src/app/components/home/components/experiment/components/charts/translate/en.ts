export const CHARTS_EN = {
    TITLE: 'Visualization',
    CHART_TITLES: {
        AVERAGE_CPU: 'Average CPU',
        AVERAGE_MEMORY: 'Average Memory',
        ERROR_RATE: 'Error Rate',
        BYTES_THROUGHPUT: 'Bytes Throughput',
        MESSAGES_THROUGHPUT: 'Messages Throughput',
        AVERAGE_TLS_HANDSHAKE_TIME: 'Average TLS Handshake Time',
    },
}
