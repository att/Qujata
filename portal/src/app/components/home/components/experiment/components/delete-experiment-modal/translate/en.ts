export const DELETE_EXPERIMENT_MODAL_EN = {
    SUBMIT_ACTION: 'Ok',
    TITLE: 'Delete Experiment',
    DESCRIPTION: 'Are you sure you want to delete "{{name}}" experiment?',
};
