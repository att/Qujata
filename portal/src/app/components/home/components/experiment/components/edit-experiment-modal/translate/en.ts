export const EDIT_EXPERIMENT_MODAL_EN = {
    SUBMIT_ACTION: 'Save',
    TITLE: 'Edit Experiment Details',
    FORM: {
        LABELS: {
            NAME: 'Experiment name',
            DESCRIPTION: 'Description',
        },
    },
};
