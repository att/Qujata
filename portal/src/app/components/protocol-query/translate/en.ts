// eslint-disable-next-line @typescript-eslint/typedef
export const PROTOCOL_QUERY_EN = {
    TITLE: 'Run new experiment',
    NOTE: {
        TITLE: 'Note:',
        TEXT: 'For each experiment you can select or type in one or more algorithms and number of iterations',
    },
    ACTION_BUTTONS: {
        RUN: 'Run',
        EXPORT: 'Export',
    },
    FIELDS_LABEL: {
        PLACEHOLDER: '+ Add new',
        EXPERIMENT_NAME: 'Experiment name',
        ITERATIONS_NUMBER: 'Number of iterations',
        ALGORITHM: 'Algorithm(s)',
        MESSAGE_SIZE: 'Message size',
        DESCRIPTION: 'Description',
    },
    FETCH_DATA: 'Loading data...',
};
  