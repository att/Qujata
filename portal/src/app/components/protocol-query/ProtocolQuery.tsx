import { noop } from 'lodash';
import React, { useCallback, useState } from 'react';
import { Options } from 'react-select';
import { ITestParams } from '../../shared/models/quantum.interface';
import { Button, ButtonActionType, ButtonSize, ButtonStyleType } from '../../shared/components/att-button';
import { AttSelect, AttSelectOption } from '../../shared/components/att-select';
import styles from './ProtocolQuery.module.scss';
import { PROTOCOL_QUERY_EN } from './translate/en';
import { Spinner, SpinnerSize } from '../../shared/components/att-spinner';
import { useGetAlgorithms, useGetIterations } from './hooks';
import { handleAlgorithmsSelection } from './utils';
import { AlgorithmsSelectorCustomOption, IterationsSelectorCustomOption } from '../../shared/components/selector-custom-option';

export type SelectOptionType = AttSelectOption | Options<AttSelectOption> | null;
type onTextChangedEvent = (e: React.ChangeEvent<HTMLInputElement>) => void;
type onTextAreaChangedEvent = (e: React.ChangeEvent<HTMLTextAreaElement>) => void;
export type OnSelectChanged = (event: SelectOptionType) => void;

export interface ProtocolQueryProps {
  isFetching: boolean;
  canExportFile?: boolean;
  onRunClick: (data: ITestParams) => void;
  onDownloadDataClicked?: () => void;
}

export const ProtocolQuery: React.FC<ProtocolQueryProps> = (props: ProtocolQueryProps) => {
  const { isFetching, canExportFile, onRunClick, onDownloadDataClicked } = props;
  const { algorithmOptions, algosBySection } = useGetAlgorithms();
  const { iterationsOptions } = useGetIterations();
  
  const [experimentName, setExperimentName] = useState('');
  const [algorithms, setAlgorithms] = useState<SelectOptionType>();
  const [prevSelectedValues, setPrevSelectedValues] = useState<string[]>([]);
  const [iterationsCount, setIterationsCount] = useState<SelectOptionType>();
  const [description, setDescription] = useState('');

  const onSubmitHandler = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    onRunClick({
      experimentName,
      algorithms: algorithms as SelectOptionType,
      iterationsCount: iterationsCount as SelectOptionType,
      description
    });
  };

  const onExperimentNameChanged: onTextChangedEvent = (event: React.ChangeEvent<HTMLInputElement>) => {
    setExperimentName(event.target.value);
  };

  const onAlgorithmsChanged: OnSelectChanged = useCallback((options: SelectOptionType): void => {
    const { newSelectedOptions, selectedValues } = 
      handleAlgorithmsSelection(options, algorithmOptions, algosBySection, prevSelectedValues);

    setAlgorithms(newSelectedOptions);
    setPrevSelectedValues(selectedValues);
  }, [algosBySection, algorithmOptions, prevSelectedValues]);

  const onIterationsNumChanged: OnSelectChanged = useCallback((options: SelectOptionType): void => {
    const selectedIterationNum: Options<AttSelectOption> = options as Options<AttSelectOption>;
    setIterationsCount(selectedIterationNum);
  }, []);

  const onDescriptionChanged: onTextAreaChangedEvent = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
    setDescription(event.target.value);
  };
  
  return (
    <div className={styles.protocol_query_wrapper}>
      <div>
        <h2 className={styles.experiment_title}>{PROTOCOL_QUERY_EN.TITLE}</h2>
        <div className={styles.note}>
          <span className={styles.note_title}>{PROTOCOL_QUERY_EN.NOTE.TITLE}</span>
          <p>{PROTOCOL_QUERY_EN.NOTE.TEXT}</p>
        </div>
      </div>
      <form className={styles.wrapper} data-testid='protocol-query-form' onSubmit={onSubmitHandler}>
          <div className={styles.form_item}>
              <label className={styles.form_item_label}>
                {PROTOCOL_QUERY_EN.FIELDS_LABEL.EXPERIMENT_NAME} <span className={styles.required}>*</span>
              </label>
              <input
                className={styles.input_form_item}
                onChange={onExperimentNameChanged}
                placeholder=''
                required
              />
          </div>
          <div className={styles.form_item}>
              <label className={styles.form_item_label}>
                {PROTOCOL_QUERY_EN.FIELDS_LABEL.ALGORITHM} <span className={styles.required}>*</span>
              </label>
              <AttSelect
                className={styles.select_form_item}
                options={algorithmOptions}
                placeholder=''
                value={algorithms as AttSelectOption[]}
                onChange={onAlgorithmsChanged}
                isMulti
                hideSelectedOptions={false}
                closeMenuOnSelect={false}
                required
                customComponent={{ Option: AlgorithmsSelectorCustomOption as React.FC }}
              />
          </div>
          <div className={styles.form_item}>
              <label className={styles.form_item_label}>
                {PROTOCOL_QUERY_EN.FIELDS_LABEL.ITERATIONS_NUMBER} <span className={styles.required}>*</span>
              </label>
              <AttSelect
                className={styles.select_form_item}
                options={iterationsOptions}
                placeholder={PROTOCOL_QUERY_EN.FIELDS_LABEL.PLACEHOLDER}
                value={iterationsCount as AttSelectOption}
                onChange={onIterationsNumChanged}
                isMulti
                hideSelectedOptions={false}
                closeMenuOnSelect={false}
                required
                customComponent={{ Option: IterationsSelectorCustomOption as React.FC }}
              />
          </div>
          <div className={styles.form_item}>
              <label className={styles.form_item_label}>
                {PROTOCOL_QUERY_EN.FIELDS_LABEL.DESCRIPTION}
              </label>
              <textarea
                className={styles.form_item_text_area}
                onChange={onDescriptionChanged}
                placeholder=''
              />
          </div>
          <div className={styles.submitButtonWrapper}>
              <Button
                disabled={isFetching}
                actionType={ButtonActionType.SUBMIT}
                size={ButtonSize.LARGE}
                styleType={ButtonStyleType.PRIMARY}
                onButtonClick={noop}
                className={styles.run_button}
              >
                {PROTOCOL_QUERY_EN.ACTION_BUTTONS.RUN}
              </Button>
              {isFetching && 
              <div role='status' className={styles.spinnerWrapper}>
                    <Spinner size={SpinnerSize.EXTRA_SMALL} />
                    <span className={styles.text}>{PROTOCOL_QUERY_EN.FETCH_DATA}</span>
              </div>}
          </div>
       </form>
       {/* <Button
            className={styles.export_button}
            actionType={ButtonActionType.BUTTON}
            size={ButtonSize.LARGE}
            styleType={ButtonStyleType.PRIMARY}
            disabled={!canExportFile}
            onButtonClick={onDownloadDataClicked}
        >
          {PROTOCOL_QUERY_EN.ACTION_BUTTONS.EXPORT}
        </Button> */}
    </div>
  );
};
