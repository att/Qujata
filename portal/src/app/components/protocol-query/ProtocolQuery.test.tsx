import '@testing-library/jest-dom';
import { RenderResult, fireEvent, render, screen, waitFor } from '@testing-library/react';
import { ProtocolQuery, ProtocolQueryProps } from './ProtocolQuery';
import { PROTOCOL_QUERY_EN } from './translate/en';

describe('ProtocolQuery', () => {
  let props: ProtocolQueryProps;
  beforeAll(() => {
    // Prepare the props for the ProtocolQuery component
    props = {
      isFetching: false,
      canExportFile: true,
      onRunClick: jest.fn(),
      onDownloadDataClicked: jest.fn(),
    };
  });

  test('should render ProtocolQuery', () => {
    // Render the ProtocolQuery component with the prepared props
    const { container }: RenderResult = render(<ProtocolQuery {...props} />);

    // Snapshot Matching
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render spinner when isFetching is true', () => {
    props.isFetching = true;
    render(<ProtocolQuery {...props} />);
    const spinner = screen.getByRole('status');
    expect(spinner).toBeInTheDocument();
  });

  test('should call onRunClick when form is submitted', () => {
    render(<ProtocolQuery {...props} />);
    const form = screen.getByTestId('protocol-query-form');
    fireEvent.submit(form);
    expect(props.onRunClick).toHaveBeenCalled();
  });

  test('should check all values were rendered', () => {
    // Render the ProtocolQuery component with the prepared props
    const { getByText }: RenderResult = render(<ProtocolQuery {...props} />);
    
    // Assert that the form elements are rendered
    expect(getByText(PROTOCOL_QUERY_EN.FIELDS_LABEL.ALGORITHM)).toBeInTheDocument();
    expect(getByText(PROTOCOL_QUERY_EN.FIELDS_LABEL.ITERATIONS_NUMBER)).toBeInTheDocument();
    expect(getByText(PROTOCOL_QUERY_EN.ACTION_BUTTONS.RUN)).toBeTruthy();
  });

  test('should handle form submission', async () => {
    render(<ProtocolQuery {...props} />);

    // Find the form and submit it
    const form = screen.getByTestId('protocol-query-form');
    fireEvent.submit(form);

    // Assert that the onRunClick function has been called
    expect(props.onRunClick).toHaveBeenCalled();
  });
});
