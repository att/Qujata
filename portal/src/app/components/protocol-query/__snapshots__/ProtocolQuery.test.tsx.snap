// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtocolQuery should render ProtocolQuery 1`] = `
<div
  class="protocol_query_wrapper"
>
  <div>
    <h2
      class="experiment_title"
    >
      Run new experiment
    </h2>
    <div
      class="note"
    >
      <span
        class="note_title"
      >
        Note:
      </span>
      <p>
        For each experiment you can select or type in one or more algorithms, number of iterations and message sizes
      </p>
    </div>
  </div>
  <form
    class="wrapper"
    data-testid="protocol-query-form"
  >
    <div
      class="form_item"
    >
      <label
        class="form_item_label"
      >
        Experiment name
         
        <span
          class="required"
        >
          *
        </span>
      </label>
      <input
        class="input_form_item"
        placeholder=""
        required=""
      />
    </div>
    <div
      class="form_item"
    >
      <label
        class="form_item_label"
      >
        Algorithm(s)
         
        <span
          class="required"
        >
          *
        </span>
      </label>
      <div
        class="att_select_container select_form_item att_select_multi_container css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class="att_select__control css-13cymwt-control"
        >
          <div
            class="att_select__value-container att_select__value-container--is-multi css-16674sh-Component"
          >
            <div
              class="att_select__placeholder css-1jqq78o-placeholder"
              id="react-select-2-placeholder"
            />
            <div
              class="att_select__input-container css-d9r47t-Component"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-2-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                aria-required="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class="att_select__input"
                id="react-select-2-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="att_select__indicators css-ny0e4k-Component"
          >
            <div
              aria-hidden="true"
              class="att_select__indicator att_select__dropdown-indicator css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
        <input
          aria-hidden="true"
          class="css-5kkxb2-requiredInput-RequiredInput"
          required=""
          tabindex="-1"
          value=""
        />
      </div>
    </div>
    <div
      class="form_item"
    >
      <label
        class="form_item_label"
      >
        Number of iterations
         
        <span
          class="required"
        >
          *
        </span>
      </label>
      <div
        class="att_select_container select_form_item att_select_multi_container css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class="att_select__control css-13cymwt-control"
        >
          <div
            class="att_select__value-container att_select__value-container--is-multi css-16674sh-Component"
          >
            <div
              class="att_select__placeholder css-1jqq78o-placeholder"
              id="react-select-3-placeholder"
            >
              + Add new
            </div>
            <div
              class="att_select__input-container css-d9r47t-Component"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-3-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                aria-required="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class="att_select__input"
                id="react-select-3-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="att_select__indicators css-ny0e4k-Component"
          >
            <div
              aria-hidden="true"
              class="att_select__indicator att_select__dropdown-indicator css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
        <input
          aria-hidden="true"
          class="css-5kkxb2-requiredInput-RequiredInput"
          required=""
          tabindex="-1"
          value=""
        />
      </div>
    </div>
    <div
      class="form_item"
    >
      <label
        class="form_item_label"
      >
        Description
      </label>
      <textarea
        class="form_item_text_area"
      />
    </div>
    <div
      class="submitButtonWrapper"
    >
      <button
        class="run_button button primary_button lg_button"
        type="submit"
      >
        Run
      </button>
    </div>
  </form>
</div>
`;
